src/main/java/de/tum/in/www1/artemis/service/connectors/ContinuousIntegrationService.java

```diff
@@ -290,13 +291,14 @@ public String forProgrammingLanguage(ProgrammingLanguage language) {
      * Name of the docker image used for the given programming language.
      *
      * @param language The programming language for which the docker image name is requested
+     * @param projectType The project type of the docker image to return
      * @return The name of the image (published on hub.docker.com)
      */
-    static String getDockerImageName(ProgrammingLanguage language) {
+    static String getDockerImageName(ProgrammingLanguage language, Optional<ProjectType> projectType) {
```

Comment:
See comment above -> may conflict with https://github.com/ls1intum/Artemis/pull/4455

Reply:
@dfuchss I think this depends on the plans of Artemis to integrate additional docker images.. If FACT will be the only one, in the near future, to have an independent image basing on `projectType`, introducing it as a `language` might be a good idea.. Otherwise this "feature/solution" will be needed anyway at some point..

Last year, for example, we got some requests about graphical Java exercises using 'Xvnc' which would eventually be integrated as `projectType`.. (not yet planned)